tests/run_tests.py

```diff
@@ -811,22 +736,116 @@ def replace_global_values(gp, replace):
+                new[key] = test_cfg
+        except KeyError:
+            raise KeyError('test configuration must specify driver')
+    return new
+
+
+def pop_run_kwargs(config):
+    '''pop run kwargs for VIC executable'''
+    run_kwargs = {}
+    run_kwargs['valgrind'] = config.pop('valgrind', False)
+    run_kwargs['mpi_proc'] = config.pop('mpi_proc', None)
+    return run_kwargs
+
+
+def check_returncode(returncode, expected=0):
+    '''check return code given by VIC, raise error if appropriate'''
+    if returncode == expected:
+        return None
+    elif returncode == default_vic_valgrind_error_code:
+        raise VICValgrindError('Valgrind raised an error')
+    else:
+        raise VICReturnCodeError('VIC return code ({0}) does not match '
+                                 'expected ({1})'.format(returncode, expected))
+
+
+def process_error(error, vic_exe):
+    '''Helper function to process possible error raised during testing'''
+    if isinstance(error, VICRuntimeError):
+        test_comment = 'Test failed during simulation'
+        error_message = error
+        tail = vic_exe.stderr
+    elif isinstance(error, VICTestError):
+        test_comment = 'Test failed during testing of output files'
+        error_message = error
+        tail = None
+    elif isinstance(error, VICValgrindError):
+        test_comment = 'Test failed do to memory error detected by valgrind'
+        error_message = error
+        tail = vic_exe.stderr
+    elif isinstance(error, VICReturnCodeError):
+        test_comment = 'Test failed due to incorrect return code'
+        error_message = error
+        tail = vic_exe.stderr
+    else:
+        raise error
+
+    print('\t{0}'.format(test_comment))
+    print('\t{0}'.format(error_message))
+    if tail is not None:
+        print('\tLast {0} lines of standard out:'.format(ERROR_TAIL))
+        print_tail(tail, n=ERROR_TAIL)
+
+    return test_comment, error_message
+
+
+def test_classic_driver_all_complete(fnames):
+    '''
+    Test that all VIC files in fnames have the same first and last index position
+    '''
+    start = None
+    end = None
+    for fname in fnames:
+        df = read_vic_ascii(fname, header=True)
+
+        # check that each dataframe includes all timestamps
+        if (start is not None) and (end is not None):
+            check_completed(df, start, end)
+        else:
+            start = df.index[0]
+            end = df.index[-1]
+
+
+def test_classic_driver_no_output_file_nans(fnames):
+    '''Test that all VIC classic driver output files in fnames have no nans'''
+    for fname in fnames:
+        df = read_vic_ascii(fname, header=True)
+        check_for_nans(df)
+
+
+def test_image_driver_no_output_file_nans(fnames, domain_file):
+    '''
+    Test that all VIC image driver output files have the same nan structure as
+    the domain file
+    '''
+    for fname in fnames:
+        ds_domain = xr.open_dataset(domain_file)
```

Comment:
Yeah, I should have caught that in your PR. You had found a bug in VIC's output files but hidden it by setting `decode_times=False`.  See my changes in `vic_init_output.c` below that address this error.